components/scream/src/share/field/field.hpp

```diff
@@ -0,0 +1,172 @@
+#ifndef SCREAM_FIELD_HPP
+#define SCREAM_FIELD_HPP
+
+#include "field_header.hpp"
+#include <share/scream_types.hpp>
+#include <share/util/scream_std_meta.hpp>
+#include <share/util/scream_kokkos_utils.hpp>
+
+#include <memory>   // For std::shared_ptr
+
+namespace scream
+{
+
+// ======================== FIELD ======================== //
+
+// A field should be composed of metadata info (the header) and a pointer to the view
+
+template<typename DataType, typename MemSpace, typename MemManagement>
+class Field {
+public:
+  using data_type           = DataType;
+  using memory_space        = MemSpace;
+  using memory_management   = MemManagement;
+  using value_type          = typename util::ScalarType<data_type>::type;
+  using view_type           = ViewType<data_type,MemSpace,MemManagement>;
+  using const_data_type     = typename view_type::traits::const_data_type;
+  using non_const_data_type = typename view_type::traits::non_const_data_type;
+  using const_field_type    = Field<const_data_type,MemSpace,MemManagement>;
+  using header_type         = FieldHeader;
+  using identifier_type     = header_type::identifier_type;
+
+  // Constructor(s)
+  Field () = delete;
+  Field (const identifier_type& id);
+  Field (std::shared_ptr<FieldHeader> fh, view_type view);
```

Comment:
Not a big deal, but `const std::shared_ptr<T>&` saves needless smart-pointer bookkeeping.